neuralprophet/forecaster.py

```diff
@@ -1381,55 +1454,74 @@ def predict(self, df, decompose=True, raw=False):
                 if periods_added > 0:
                     fcst = fcst[:-periods_added]
             df_list_predict.append(fcst)
-        df = df_list_predict[0] if len(df_list_predict) == 1 else df_list_predict
+        if df_names is not None:
+            df = dict(zip(df_names, df_list_predict))
+        else:
+            df = df_list_predict[0] if len(df_list_predict) == 1 else df_list_predict
         return df
 
-    def _predict_trend(self, df):
+    def _predict_trend(self, df, df_names=None):
         """Predict only trend component of the model.
 
         Args:
             df (pd.DataFrame): containing column 'ds', prediction dates
-
+            df_names (list): list of names of dataframes provided (used for local modeling or local normalization)
         Returns:
             pd.Dataframe with trend on prediction dates.
 
         """
         df = self._check_dataframe(df, check_y=False, exogenous=False)
-        df = df_utils.normalize(df, self.data_params, local_modeling=self.local_modeling)
+        df = df_utils.normalize(df, self.data_params, local_modeling=self.local_modeling, df_names=df_names)
         t = torch.from_numpy(np.expand_dims(df["t"].values, 1))
         trend = self.model.trend(t).squeeze().detach().numpy()
-        trend = trend * self.data_params["y"].scale + self.data_params["y"].shift
+        if self.local_modeling:
+            scale_y, shift_y = (
+                self.data_params.norm_params_dict[df_names]["y"].scale,
+                self.data_params.norm_params_dict[df_names]["y"].shift,
+            )
+        else:
+            scale_y, shift_y = self.data_params["y"].scale, self.data_params["y"].shift
+        trend = trend * scale_y + shift_y
         return pd.DataFrame({"ds": df["ds"], "trend": trend})
 
     def predict_trend(self, df):
         """Predict only trend component of the model.
 
         Args:
             df (pd.DataFrame): containing column 'ds', prediction dates
-
         Returns:
             pd.Dataframe or list of pd.Dataframe with trend on prediction dates.
 
         """
-        df_list = df_utils.create_df_list(df)
+        df_list, df_names = df_utils.create_df_list(df)
+        if self.local_modeling:
+            df_utils.check_compatibility_df_names(self.df_names, df_names, df_list)
+            list_of_names = df_names
+        else:
+            if df_names is not None:
+                log.info("dict of DataFrames provided - Ignoring keys as local modeling is not being used")
+            list_of_names = [None] * len(df_list)
```

Comment:
use `else` to only create a None-list if df_names not None, otherwise keep names

Reply:
df_names can be not None if a dict is passed but local_modeling is not being used. Therefore, a None-list is required. df_names is later used for the creation of the output dict.